dump/plugins/route.py

```diff
@@ -0,0 +1,296 @@
+import json
+import re
+import copy
+from dump.match_infra import MatchEngine, MatchRequest
+from dump.helper import create_template_dict, verbose_print
+from .executor import Executor
+
+NH = "ASIC_STATE:SAI_OBJECT_TYPE_NEXT_HOP"
+NH_GRP = "ASIC_STATE:SAI_OBJECT_TYPE_NEXT_HOP_GROUP"
+RIF = "ASIC_STATE:SAI_OBJECT_TYPE_ROUTER_INTERFACE"
+CPU_PORT = "ASIC_STATE:SAI_OBJECT_TYPE_PORT"
+
+OID_HEADERS = {
+    NH: "0x40",
+    NH_GRP: "0x50",
+    RIF: "0x60",
+    CPU_PORT: "0x10"
+}
+
+
+def get_route_pattern(dest):
+    return "*\"dest\":\"" + dest + "\"*"
+
+
+def get_vr_oid(asic_route_entry):
+    matches = re.findall(r"\{.*\}", asic_route_entry)
+    key_dict = {}
+    if matches:
+        try:
+            key_dict = json.loads(matches[0])
+        except Exception as e:
+            pass
+    return key_dict.get("vr", "")
+
+
+class NextHopGroupMatchOptimizer():
+    """
+    A Stateful Wrapper which optimizes the the queries by caching the redis keys.
+    Will be used for these ASIC keys in particular
+    1) SAI_OBJECT_TYPE_NEXT_HOP_GROUP_MEMBER
+    2) SAI_OBJECT_TYPE_NEXT_HOP
+    3) SAI_OBJECT_TYPE_ROUTER_INTERFACE
+    Note: Caches all the kv pairs for a key
+    """
+
+    def __init__(self, m_engine):
+        self.key_cache = dict()
```

Comment:
Suggest to use one way to initialize dictionary.
In some places it is ```dict()``` and in some ```{}```, e.g. line 27.
(BTW, ```{}``` is a bit faster comparing to ```dict()```)